File: Data Structures & Algorithms/duplicate-integer/submission-2.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:

        # understand the problem 
        # we are given an array of numbers, we need to return true if there is a duplicate in the list

        # implement solution

        # declare some variable that holds our value, true or false, we wll return this later
        isDup = False
       
        # create a dictionary to hold each letter; key, and its occurence: value
        numDictionary = {}

        # iterate through the list and add into dictionary 
        for num in nums:
            numDictionary[num] = numDictionary.get(num, 0) + 1

        # finally go thru the dictionaries values and if there is a value greater than 1, return true
        for key, value in numDictionary.items():
            if value != 1:
                isDup = True
        
        return isDup

        

File: Data Structures & Algorithms/two-integer-sum/submission-2.java
class Solution {
    public int[] twoSum(int[] nums, int target) {

        // we will return an array of the 2 indexes that add up to the target
        int[] indexs = new int[2];

        // iterate thru the array 
        for(int i = 0; i < nums.length; i++){
            for(int j = 0; j < nums.length; j++){
                // if nums i and nums j = target and i and j are not the same
               if(nums[i] + nums[j] == target && i != j) {
                if(i < j){
                    indexs[0] = i;
                    indexs[1] = j; 
                } else {
                    indexs[1] = i;
                    indexs[0] = j; 
                }
            }
        }
     }

            

        return indexs; 
    }
}
